File: backend/src/main/java/com/gxs/bank/controller/KycController.java
package com.gxs.bank.controller;

import com.gxs.bank.dto.response.ApiResponse;
import com.gxs.bank.model.KycDocument;
import com.gxs.bank.model.User;
import com.gxs.bank.service.KycService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/kyc")
@RequiredArgsConstructor
public class KycController {
    private final KycService kycService;

    @PostMapping("/submit")
    public ResponseEntity<ApiResponse<KycDocument>> submitKyc(@AuthenticationPrincipal User user, @RequestBody Map<String, String> request) {
        KycDocument.DocumentType type = KycDocument.DocumentType.valueOf(request.get("documentType").toUpperCase());
        String documentNumber = request.get("documentNumber");
        String fileUrl = request.getOrDefault("fileUrl", "dummy_url");
        
        KycDocument doc = kycService.submitDocument(user.getId(), type, documentNumber, fileUrl);
        return ResponseEntity.ok(ApiResponse.ok("KYC submitted successfully", doc));
    }

    @GetMapping("/documents")
    public ResponseEntity<ApiResponse<List<KycDocument>>> getDocuments(@AuthenticationPrincipal User user) {
        return ResponseEntity.ok(ApiResponse.ok(kycService.getUserDocuments(user.getId())));
    }
}

File: backend-python/com/gxs/bank/controller/KycController.py
from fastapi import APIRouter, Depends

from com.gxs.bank.config.database import get_db
from com.gxs.bank.config.serialization import serialize
from com.gxs.bank.dto.response.ApiResponse import ok
from com.gxs.bank.model.KycDocument import KycDocument
from com.gxs.bank.security.JwtAuthenticationFilter import get_current_user
from com.gxs.bank.service.KycService import KycService


router = APIRouter(prefix="/api/kyc", tags=["KYC"])


@router.post("/submit")
def submitKyc(payload: dict, current_user=Depends(get_current_user), db=Depends(get_db)):
    document_type = KycDocument.DocumentType(payload.get("documentType", "").upper())
    document_number = payload.get("documentNumber")
    file_url = payload.get("fileUrl", "dummy_url")

    doc = KycService(db).submitDocument(current_user.id, document_type, document_number, file_url)
    return ok(serialize(doc), "KYC submitted successfully")


@router.get("/documents")
def getDocuments(current_user=Depends(get_current_user), db=Depends(get_db)):
    docs = KycService(db).getUserDocuments(current_user.id)
    return ok(serialize(docs))
